lib/plugins/aws/deploy/lib/mergeIamTemplates.js

```diff
@@ -123,24 +136,34 @@ module.exports = {
         });
       }
 
-      // add custom iam role statements
-      if (this.serverless.service.provider.iamRoleStatements &&
-        this.serverless.service.provider.iamRoleStatements instanceof Array) {
-        this.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources
-          .IamPolicyLambdaExecution
-          .Properties
-          .PolicyDocument
-          .Statement = this.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources
-          .IamPolicyLambdaExecution
-          .Properties
-          .PolicyDocument
-          .Statement.concat(this.serverless.service.provider.iamRoleStatements);
+      if (this.serverless.service.provider.iamRoleStatements) {
+        const reason = checkStatements(this.serverless.service.provider.iamRoleStatements);
```

Comment:
n/a given that now the checkStatements (which was renamed to validateStatemenets) throws the error itself (instead of returning an error message)